custom-login/src/components/AuthRequiredModal.vue

```diff
@@ -0,0 +1,57 @@
+<template>
+  <div>
+  <ConfirmModal
+    v-model="isModalVisible"
+    @confirm=confirmModal
+    @cancel=closeModal
+  >
+    <template v-slot:title>Auth requried</template>
+    <template v-slot:confirm>Yes</template>
+    <template v-slot:cancel>No</template>
+    <p>Do you want to re-authenticate?</p>
+  </ConfirmModal>
+  </div>
+</template>
+
+<script>
+import ConfirmModal from './ConfirmModal.vue';
+import router from '../router';
+import { emitter } from '../main';
+export default {
+  name: 'AuthRequiredModal',
+  data() {
+    return {
+      isModalVisible: false,
+      oktaAuth: null
+    }
+  },
+  created() {
+    emitter.on('_onAuthRequired', (oktaAuth) => {
+      this.oktaAuth = oktaAuth;
+      if (!oktaAuth.authStateManager.getPreviousAuthState()?.isAuthenticated) {
+        // App initialization stage
+        this.confirmModal();
```

Comment:
How about call signInWithRedirect directly instead of wrapping it in the modal logic?

Reply:
I think this is the "custom-login" sample which means it shows a SIW on the /login route

Reply:
Done.  `this.confirmModal();`  replaced to `router.push('/login');` for custom-login sample and `this.oktaAuth.signInWithRedirect();` for okta-hosted-login sample